src/pybind/mgr/dashboard/controllers/telemetry.py

```diff
@@ -0,0 +1,34 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+
+from . import ApiController, RESTController
+from .. import mgr
+from ..exceptions import DashboardException
+from ..security import Scope
+
+
+@ApiController('/telemetry', Scope.CONFIG_OPT)
+class Telemetry(RESTController):
+
+    @RESTController.Collection('GET')
+    def report(self):
+        return mgr.remote('telemetry', 'compile_report')
+
+    @RESTController.Collection('GET')
+    def device_report(self):
+        return mgr.remote('telemetry', 'gather_device_report')
```

Comment:
I just haven't thought about it but I agree in case of the dashboard we can merge both reports into a single endpoint/file.

I don't think it would be a good idea to merge them into one within the telemetry module as both report are going to be send to different endpoints (by default): https://telemetry.ceph.com/report and https://telemetry.ceph.com/device .

Reply:
Telemetry has 2 CLI commands to display reports kind of the same way we do here: `telemetry show` and `telemetry show-device` (I do think they should be better named `show report` and `show device`), but you can do the same you did for `on` and `off` methods and add a new telemetry command (`show-all`) and generate the same output that you are generating in the dashboard. 

That way, every time someone adds more reports hopefully they'll not forget to add those to the `show-all` CLI command. My point is that: there's no easy way for dashboard to enforce dependencies on other components, but CLI API is (more or less) always respected, so let's pin our dependencies to the CLI ones.

Reply:
Yes, I understand. Thank you for the explanation. I'll try to adapt the code as suggested.